modules/core/03-connection/keeper/handshake.go

```diff
@@ -242,6 +247,10 @@ func (k Keeper) ConnOpenAck(
 	connection.Counterparty.ConnectionId = counterpartyConnectionID
 	k.SetConnection(ctx, connectionID, connection)
 
+	// delete generatedConnectionID mapping now that handshake attempt is successful
+	// we no longer need to store it for redundancy protection
+	k.DeleteGeneratedConnectionID(ctx, connection.ClientId, counterpartyConnectionID)
```

Comment:
Not sure if I should be deleting in ACK and CONFIRM. No longer strictly necessary for the state machine but may be useful as a query even after handshake is complete

Reply:
Shouldn't we only need to delete in the `ConnOpenConfirm` handler?

Reply:
Well, theoretically we could have a crossing hello; where ACK gets called on both ends

Reply:
In this case does CONFIRM not get called on both ends too?

Reply:
nope because both sides go to OPEN